build/Microsoft.DotNet.Cli.Prepare.targets

```diff
@@ -182,29 +182,77 @@
                                OverwriteDestination="True" />
   </Target>
 
-  <Target Name="RestorePackages"
-          DependsOnTargets="SetupRestorePackagesInputsOutputs;Init"
-          Inputs="@(RestorePackagesInput)"
-          Outputs="@(RestorePackagesInput->'%(RelativeDir)/project.lock.json')">
+  <Target Name="RestoreSrcPackages"
+          DependsOnTargets="SetupRestoreSrcPackagesInputsOutputs;Init"
+          Inputs="@(RestoreSrcPackagesInput)"
+          Outputs="@(RestoreSrcPackagesInput->'%(RelativeDir)/obj/project.assets.json');@(RestoreSrcPackagesInput->'%(RelativeDir)/obj/%(Filename).csproj.nuget.g.props');@(RestoreSrcPackagesInput->'%(RelativeDir)/obj/%(Filename).csproj.nuget.g.targets')">
+
+    <CallTarget Targets="CleanSrcLockFiles" />
+
+    <DotNetRestoreMsbuildLegacy WorkingDirectory="%(RestoreSrcPackagesInput.RootDir)%(RestoreSrcPackagesInput.Directory)" ToolPath="$(DotNetPath)" Root="&quot;%(RestoreSrcPackagesInput.FullPath)&quot;" />
+
+  </Target>
+
+  <Target Name="CleanSrcLockFiles" >
+    <ItemGroup>
+      <SrcLockFiles Include="$(RepoRoot)/src/**/project.assets.json;$(RepoRoot)/src/**/*.csproj.nuget.g.props;$(RepoRoot)/src/**/*.csproj.nuget.g.targets" />
+    </ItemGroup>
+    <Delete Files="@(SrcLockFiles)" />
+  </Target>
+
+  <Target Name="SetupRestoreSrcPackagesInputsOutputs">
+    <ItemGroup>
+      <RestoreSrcPackagesInput Include="$(RepoRoot)/src/**/*.csproj" Exclude="$(RepoRoot)/src/**/%24projectName%24.csproj"/>
+    </ItemGroup>
+  </Target>
+
+  <Target Name="RestoreToolsPackages"
+          DependsOnTargets="SetupRestoreToolsPackagesInputsOutputs;Init"
+          Inputs="@(RestoreToolsPackagesInput)"
+          Outputs="@(RestoreToolsPackagesInput->'%(RelativeDir)/obj/project.assets.json');@(RestoreToolsPackagesInput->'%(RelativeDir)/obj/%(Filename).csproj.nuget.g.props');@(RestoreToolsPackagesInput->'%(RelativeDir)/obj/%(Filename).csproj.nuget.g.targets')">
+
+    <CallTarget Targets="CleanToolsLockFiles" />
+
+    <DotNetRestoreMsbuildLegacy WorkingDirectory="%(RestoreToolsPackagesInput.RootDir)%(RestoreToolsPackagesInput.Directory)" ToolPath="$(DotNetPath)" Root="&quot;%(RestoreToolsPackagesInput.FullPath)&quot;" />
+
+  </Target>
+
+  <Target Name="CleanToolsLockFiles" >
+    <ItemGroup>
+      <ToolsLockFiles Include="$(RepoRoot)/tools/**/project.assets.json;$(RepoRoot)/tools/**/*.csproj.nuget.g.props;$(RepoRoot)/tools/**/*.csproj.nuget.g.targets" />
+    </ItemGroup>
+    <Delete Files="@(ToolsLockFiles)" />
+  </Target>
+
+  <Target Name="SetupRestoreToolsPackagesInputsOutputs">
```

Comment:
do we need to do this restore so far out? It would be nice to capture this restore in the context of those projects.